riscv-executor/src/lib.rs

```diff
@@ -707,27 +1109,51 @@ impl<'a, 'b, F: FieldElement> Executor<'a, 'b, F> {
                 let val = self.proc.get_mem(addr as u32 & 0xfffffffc);
                 let rem = addr % 4;
 
-                self.proc.set_reg_mem(write_addr1, val.into());
-                self.proc.set_reg_mem(write_addr2, rem.into());
+                self.reg_write(2, write_addr1, val.into(), 3);
+                self.reg_write(3, write_addr2, rem.into(), 4);
+
+                set_col!(tmp1_col, addr1);
+                set_col!(tmp3_col, Elem::from_u32_as_fe(val));
+                set_col!(tmp4_col, Elem::from_i64_as_fe(rem));
+
+                let v = addr1.add(&args[1]).as_i64_from_lower_bytes();
+                let (b1, b2, b3, b4, _sign) = decompose_lower32(v);
+                set_col!(X_b1, Elem::from_u32_as_fe((b1 / 4).into()));
+                set_col!(X_b2, Elem::from_u32_as_fe(b2.into()));
+                set_col!(X_b3, Elem::from_u32_as_fe(b3.into()));
+                set_col!(X_b4, Elem::from_u32_as_fe(b4.into()));
+                set_col!(
+                    wrap_bit,
+                    Elem::from_u32_as_fe(((v as u64 >> 32) & 1) as u32)
+                );
+
+                self.proc.memory_machine.read(
+                    self.step + 1,
+                    addr as u32 & 0xfffffffc,
+                    val.into(),
+                    0,
+                );
 
                 Vec::new()
             }
+            // TODO: update to witness generation for continuations
```

Comment:
what's missing here for continuations?

Reply:
I think the instructions are correct actually. Ah I guess maybe missing the columns?